Fix Protosynthesis and Quark Drive boosts

AlexOn1ine opened the pull request:
Currently both Atk/Sp.Atk and Def/Sp.Def are boosted if either of them is the highest stat.
This should fix it.

## Images
Protosynthesis Nidoqueen.
Before:
![pokeemerald-15](https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/ea3e878f-9e8c-4fd1-b1e0-afc633cec229)
After:
![pokeemerald-14](https://github.com/rh-hideout/pokeemerald-expansion/assets/93446519/e659832e-c3d9-4ba7-92ec-f62b46fe6023)

## **Discord contact info**
rainonline

LOuroboros commented on src/battle_util.c:
```diff
@@ -8923,18 +8925,19 @@ static u32 CalcMoveBasePowerAfterModifiers(u16 move, u8 battlerAtk, u8 battlerDe
+            && ((IS_MOVE_PHYSICAL(move) && atkHighestStat == STAT_ATK) || (IS_MOVE_SPECIAL(move) && atkHighestStat == STAT_SPATK)))
                 MulModifier(&modifier, UQ_4_12(0.7));
         }
-        break;
```
Please re-add that `break;`.

AlexOn1ine replied:
Oh...

Fixed it.

DizzyEggg commented:
I'd like to see a test added for this 👀

AlexOn1ine commented:
Added tests and fixed a bug I introduced. The Tests fail without this PR.

Also might be a good Idea to replace the mons once gen9 species are merged.
